File: frontend/src/types/material.ts
export interface Material {
  id: string;
  subject_id: string | null;
  filename: string;
  original_filename: string;
  file_type: string;
  file_size: number | null;
  processing_status: 'uploaded' | 'processing' | 'processed' | 'failed';
  created_at: string;
  updated_at: string;
}

export interface MaterialUploadResponse {
  id: string;
  subject_id: string | null;
  filename: string;
  original_filename: string;
  file_type: string;
  file_size: number | null;
  processing_status: string;
  created_at: string;
  updated_at: string;
}

export interface ApiError {
  detail: string;
}

File: backend/app/schemas/material.py
"""Pydantic schemas for the Material API."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class MaterialCreate(BaseModel):
    """Payload for creating a material (used internally by service)."""

    subject_id: uuid.UUID


class MaterialResponse(BaseModel):
    """Shape of a material returned by the API."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    subject_id: uuid.UUID | None
    filename: str
    original_filename: str
    file_type: str
    file_size: int | None
    processing_status: str
    created_at: datetime
    updated_at: datetime